packages/core/src/modules/credentials/protocol/v2/CredentialMessageBuilder.ts

```diff
@@ -119,46 +114,38 @@ export class CredentialMessageBuilder {
     credentialRecord: CredentialExchangeRecord,
     options: ServiceOfferCredentialOptions
   ): Promise<V2OfferCredentialMessage> {
-    if (formatServices.length === 0) {
-      throw new AriesFrameworkError('no format services provided to createProposal')
-    }
+    if (formatServices.length === 0)
+      throw new AriesFrameworkError('no format services provided to createOfferAsResponse')
+
     // create message
     // there are two arrays in each message, one for formats the other for attachments
-    const formatsArray: CredentialFormatSpec[] = []
-    const offersAttachArray: Attachment[] | undefined = []
-    let previewAttachments: V2CredentialPreview = new V2CredentialPreview({
-      attributes: [],
-    })
+    const formats: CredentialFormatSpec[] = []
+    const offerAttachments: Attachment[] = []
+    const credentialPreview = new V2CredentialPreview({ attributes: [] })
 
     for (const formatService of formatServices) {
-      const { attachment: offersAttach, preview, format } = await formatService.createOffer(options)
-      if (offersAttach) {
-        offersAttachArray.push(offersAttach)
-      } else {
-        throw new AriesFrameworkError('offersAttach not initialized for credential proposal')
-      }
-      if (preview && preview.attributes.length > 0) {
-        previewAttachments = preview
-      }
-      formatsArray.push(format)
+      const { attachment, preview, format } = await formatService.createOffer(options)
+
+      if (preview && preview.attributes.length > 0) credentialPreview.attributes = preview.attributes
 
-      await formatService.processOffer(offersAttach, credentialRecord)
+      formats.push(format)
+      offerAttachments.push(attachment)
+
+      await formatService.processOffer(attachment, credentialRecord)
     }
 
     const messageProps: V2OfferCredentialMessageOptions = {
       id: this.generateId(),
-      formats: formatsArray,
+      formats,
       comment: options.comment,
-      offerAttachments: offersAttachArray,
-      credentialPreview: previewAttachments,
+      offerAttachments,
+      credentialPreview,
     }
-    const credentialOfferMessage: V2OfferCredentialMessage = new V2OfferCredentialMessage(messageProps)
+    const credentialOfferMessage = new V2OfferCredentialMessage(messageProps)
 
-    credentialOfferMessage.setThread({
-      threadId: credentialRecord.threadId,
-    })
+    credentialOfferMessage.setThread(credentialRecord)
```

Comment:
Not sure about this one, would like to keep it like this:

```suggestion
    credentialOfferMessage.setThread({ threadId: credentialRecord.threadId })
```